### DISCUSSION TITLE
Is there going to be any EF Core ICollection<> support going forward with 11+?

### DISCUSSION THREAD

**Discussion by replaysMike** [LuckyPennySoftware/AutoMapper#3894] [NA]
Currently there doesn't appear to be any solutions for EF Core mapping onto existing collections, as neither [AutoMapper.Collections](https://github.com/AutoMapper/AutoMapper.Collection) or [AutoMapper.Collection.EntityFrameworkCore](https://github.com/AutoMapper/AutoMapper.Collection.EFCore) doesn't support v11 and no word on if it ever will.

Situations such as the following don't work:
```csharp
var entity = _mapper.Map<MyModel, MyEntity>(model, entity);
```
and require difficult `AfterMap()` mapping behaviors configured which are a pain:
```csharp
CreateMap<MyModel, DataModel.MyEntity>()
  .AfterMap(DontOverwriteMyCollections);

private void DontOverwriteMyCollections(MyModel from, DataModel.MyEntity to) 
{
  to.Id = from.Id;
  var i = 0;
  // no static Mapper.Map<>() available anymore, so this is tedious
  foreach(var row in from.Relationship) {
    var dest = to.Relationship.Skip(i).First();
    dest.SubId = row.SubId;
    // etc
  }
}
```

is there a.. better way?

**Comment by jbogard** [comment]
Open a PR for those two to support v11 and latest EF Core?
